bin/PerlDDS/Run_Test.pm

```diff
@@ -376,6 +376,8 @@ sub new {
   my $class = shift;
   my $self = bless {}, $class;
 
+  $self->{configs} = shift;
+  $self->{config} = shift;
```

Comment:
I don't know if Perl has a term for this, but if these are optional consider making these "keyword" arguments as Python calls them. This way they are explicit and it makes it much easier to add more arguments in the future in a backwards compatible way. The idea is because arrays and hashes are basically the same thing in Perl you can do:

```perl
my $test = new PerlDDS::TestFramework(configs => \%configs, config => 'ir_tcp');
```

Then treat `@_` as a hash that can be joined with a default argument hash.

Here's an example of this I've implemented before:

https://github.com/OpenDDS/OpenDDS/blob/07ee200864c1b58616cf5e3ab3dc865fb71d84ca/tools/scripts/modules/command_utils.pm#L264-L282

Here I'd recommend to something like:

```perl
# At the top we'd need to import trace since we're currently
# not using in the test framework, but really should:
use FindBin;
use lib "$FindBin::RealBin/../../tools/scripts/modules"
use misc_utils qw(trace);

# ...

  # At where I'm commenting replace the bless, configs, and config lines with:
  my %valid_args = (
    configs => {},
    config => undef,
  );

  my %args = (%valid_args, @_);
  my @invalid_args = grep { !exists($valid_args{$_}) } keys(%args);
  trace("invalid arguments: ", join(', ', @invalid_args)) if (scalar(@invalid_args));
  my $self = bless(\%args, $class);
```

Because `args` is the basis of the `bless`, `configs` and `config` will be in `$self`. We could also move most or all of the other values below into the `valid_args` hash to allow overriding them from the constructor.